### PR TITLE
 Adding workflow for automatic testing

### PR DESCRIPTION
**Pull request by alexdesiqueira**
Here we add a workflow on GHA to trigger automatic tests.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Running tests
+on: [push, pull_request]
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    env:
+      OS: ubuntu-latest
+      PYTHON: '3.9'
```
Comment (stefanv) [inline .github/workflows/tests.yml:8]: This says 3.9, but later it uses 3.7?

Comment (alexdesiqueira) [reply .github/workflows/tests.yml:8]: Thanks for pointing that out, I'll setup the same machine we use for tests instead.

Comment (alexdesiqueira) [reply .github/workflows/tests.yml:8]: Thanks for pointing that out, I copied this almost verbatim from the other workflow (which was copied from the CodeCov example :sweat_smile:)
I'll check how to set a proper machine and fix this.

**Comment by stefanv** [comment]
I guess I was mostly confused about why the Python version is specified twice and is different. Maybe it doesn't matter, but then a comment would be great.

This looks good! Go ahead and merge when ready.

**Comment by alexdesiqueira** [comment]
@stefanv this was mostly rewritten, and I think that for the best :slightly_smiling_face:
Merging.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (NeMeCChip) [reply .github/workflows/tests.yml]: _[]()_

**Pull request merged**